scripts/build_demand_profiles.py

```diff
@@ -0,0 +1,207 @@
+        scale:
+        ssp:
+        weather_year:
+        prediction_year:
+        region_load:
+
+        
+Inputs
+------
+
+- ``networks/base.nc``: confer :ref:`base`, a base Pypsa Network 
+- ``resources/bus_regions/regions_onshore.geojson``: confer :ref:`bus regions`
+- ``resources/ssp2-2.6/2030/era5_2013/Africa.nc``: hourly country load profiles produced by GEGIS
+- ``load_data_paths``
+- ``resources/shapes/gadm_shapes.geojson``: confer :ref:`shapes`, file containing the gadm shapes
+
+Outputs
+-------
+
+- ``resources/demand_profiles.csv``: the content of the file is the electric demand profile associated to each bus. The file has the snapshots as rows and the buses of the network as columns. 
+
+Description
+-----------
+
+The rule :mod:`build_demand` creates load demand profiles in correspondance of the buses of the network.
+It creates the load paths for GEGIS outputs by combining the input parameters of the countries, weather year, prediction year, and SSP scenario.
+Then with a function that takes in the PyPSA network "base.nc", region and gadm shape data, the countries of interest, a scale factor, and the snapshots,
+it returns a csv file called "demand_profiles.csv", that allocates the load to the buses of the network according to GDP and population.
+
+"""
+
+import logging
+import os
+
+import geopandas as gpd
+import numpy as np
+import pandas as pd
+import powerplantmatching as pm
+import pypsa
+import xarray as xr
+from _helpers import configure_logging, getContinent, update_p_nom_max
+from shapely.validation import make_valid
+from vresutils import transfer as vtransfer
+
+logger = logging.getLogger(__name__)
+
+
+def normed(s):
+    return s / s.sum()
+
+
+def get_load_paths_gegis(ssp_parentfolder, config):
+    """
+    Creates load paths for the GEGIS outputs
+
+    The paths are created automatically according to included country,
+    weather year, prediction year and ssp scenario
+
+    Example
+    -------
+    ["/data/ssp2-2.6/2030/era5_2013/Africa.nc", "/data/ssp2-2.6/2030/era5_2013/Africa.nc"]
+    """
+    countries = config.get("countries")
+    region_load = getContinent(countries)
+    weather_year = config.get("load_options")["weather_year"]
+    prediction_year = config.get("load_options")["prediction_year"]
+    ssp = config.get("load_options")["ssp"]
+
+    load_paths = []
+    for continent in region_load:
+        load_path = os.path.join(
+            ssp_parentfolder,
+            str(ssp),
+            str(prediction_year),
+            "era5_" + str(weather_year),
+            str(continent) + ".nc",
+        )
+        load_paths.append(load_path)
+
+    return load_paths
+
+
+def build_demand_profiles(
+    n,
+    load_paths,
+    regions,
+    admin_shapes,
+    countries,
+    scale,
+    start_date,
+    end_date,
+):
+    """
+    Add load to the network and distributes them according GDP and population.
+
+    Parameters
+    ----------
+    n : pypsa network
+    regions : .geojson
```

Comment:
change the order of the description to align with the read in above (line 94-101)

Reply:
done